core/domain/draft_upgrade_services.py

```diff
@@ -87,6 +87,80 @@ def try_upgrading_draft_to_exp_version(
     return draft_change_list
 
 
+def extract_html_from_draft_change_list(draft_change_list):
+    """Extracts all the HTML strings from a draft change list.
+
+    Args:
+        draft_change_list: list(ExplorationChange). The list of
+            ExplorationChange domain objects to upgrade.
+
+    Returns:
+        list(str). The list of html content strings.
+    """
+    html_list = []
+    for change in draft_change_list:
+        new_value = None
+        if not change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY:
+            continue
+        # The change object has the key 'new_value' only if the
+        # cmd is 'CMD_EDIT_STATE_PROPERTY' or
+        # 'CMD_EDIT_EXPLORATION_PROPERTY'.
+        new_value = change.new_value
+        if change.property_name == exp_domain.STATE_PROPERTY_CONTENT:
+            # new_value['html'] = conversion_fn(new_value['html'])
+            html_list += [new_value['html']]
+        elif (change.property_name ==
+              exp_domain.STATE_PROPERTY_INTERACTION_CUST_ARGS):
+            # Only customization args with the key 'choices' have HTML
+            # content in them.
+            if 'choices' in new_value.keys():
+                html_list += new_value['choices']['value']
+        elif (change.property_name ==
+              exp_domain.STATE_PROPERTY_WRITTEN_TRANSLATIONS):
+            written_translations_object = (
+                state_domain.WrittenTranslations.from_dict(new_value))
+            html_in_written_translations = (
+                written_translations_object.get_all_html_content_strings())
+            html_list += html_in_written_translations
```

Comment:
`extract_latex_values_from_math_rich_text_without_filename`  will extract the latex value only from the old hint in the change list. this is because for the new hint the `svg_filename` field won't be empty.
So finally we will be generating image only for the old hints with empty svg_filename field.